sdk/opendp/whitenoise/synthesizers/dpctgan/dpctgan.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+import torch
+from torch import optim
+from torch.nn import functional
+import torch.nn as nn
+import torch.utils.data
+from torch.nn import BatchNorm1d, Dropout, LeakyReLU, Linear, Module, ReLU, Sequential,Sigmoid
+from torch.nn import functional as F
+from opendp.whitenoise.synthesizers.base import SDGYMBaseSynthesizer
+
+import ctgan
+from ctgan.transformer import DataTransformer
+from ctgan.conditional import ConditionalGenerator 
+from ctgan.models import Discriminator, Generator
+from ctgan.sampler import Sampler 
+
+from ctgan import CTGANSynthesizer
+
+import torchdp
```

Comment:
We should move this into a torchdp directory, and have that directory pip install torchdp

 sdk/opendp/whitenoise/synthesizers/torchdp/

@anujdesai510  since you will probably need to do something similar, this can be how we deal with the optional dependency for now